lumen/ai/agents.py

```diff
@@ -157,14 +157,31 @@ def _get_schema(self, source: Source | Pipeline, table: str = None):
         schema = dict(schema)
         for field, spec in schema.items():
             if "enum" in spec:
-                schema[field] = dict(spec, enum=spec["enum"][:5])
+                spec.pop("enum")
+            if "inclusiveMinimum" in spec:
+                spec["min"] = spec.pop("inclusiveMinimum")
+            if "inclusiveMaximum" in spec:
+                spec["max"] = spec.pop("inclusiveMaximum")
         return schema
 
+
+    async def _get_closest_tables(self, messages: list | str, tables: list[str], n: int = 3) -> list[str]:
+        table_keywords = (await self.llm.invoke(
+            messages,
+            system=self.system_prompt,
+            response_model=FuzzyTable,
+            allow_partial=False
+        )).keywords
```

Comment:
Doesn't this need some actual instructions to extract the keywords most relevant to the table selection? The description of the keywords parameter on the FuzzyTable model does not really seem sufficient.

Reply:
Seems like instead of giving it the system_prompt it should be a prompt tailored to keyword extraction.